File: scm-csharp/scheme/primitives/PrimitiveCurrentDirectory.cs
namespace scheme;

public class PrimitiveCurrentDirectory : Primitive
{
    public override string Name()
    {
        return "current-directory";
    }

    public override string Info()
    {
        return
            "Syntax: (current-directory)\n" +
            "Library: (scm system)\n" +
            "Description: Returns the current working directory as a string.\n" +
            "Example:\n" +
            "  (current-directory) => \"/home/user/projects\"";
    }
    
    public override object Apply(SourcePos? pos, object[] arguments)
    {
        CheckArgs(pos, arguments, 0, 0);
        return Directory.GetCurrentDirectory().ToCharArray();
    }
}

File: scm-java/src/main/scheme/primitives/PrimitiveCurrentDirectory.java
package scheme.primitives;

import scheme.*;

public class PrimitiveCurrentDirectory extends Primitive {
    @Override
    public String name() {
        return "current-directory";
    }

    @Override
    public String info() {
        return "Syntax: (current-directory)\n" +
               "Library: (scm fs)\n" +
               "Description: Returns the current working directory as a string.\n" +
               "Example:\n" +
               "  (current-directory) => \"/home/user/projects\"";
    }
    
    @Override
    public Object apply(SourcePos pos, Object[] arguments) {
        checkArgs(pos, arguments, 0, 0);
        return System.getProperty("user.dir").toCharArray();
    }
}
